client/my-sites/controller.js

```diff
@@ -112,46 +111,62 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
-function isPathAllowedForDomainOnlySite( pathname, domainName ) {
+function renderSelectedSiteIsDomainOnly( { reactContext, selectedSite } ) {
+	const EmptyContentComponent = require( 'components/empty-content' );
+	const { store: reduxStore } = reactContext;
+
+	removeSidebar( reactContext );
+
+	renderWithReduxStore(
+		React.createElement( EmptyContentComponent, {
+			title: i18n.translate( 'This feature is not available for domains' ),
+			line: i18n.translate( 'To use this feature you need to create a site' ),
+			action: i18n.translate( 'Create New Site' ),
+			actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
+			secondaryAction: i18n.translate( 'Domain Management' ),
+			secondaryActionURL: domainManagementList( selectedSite.slug )
+		} ),
+		document.getElementById( 'primary' ),
+		reduxStore
+	);
+}
+
+function isPathAllowedForDomainOnlySite( { reactContext: { pathname }, selectedSite } ) {
```

Comment:
This is subject to debate I think. This syntax allows us to mimic named parameters which are extremely useful  but in the same time, grouping different parameters into an "option" object just for the sake of it makes it harder to read.
In this case I would agree with Stéphane, there are only 2 parameters for all theses functions: a context and a selectedSite, I don't think it helps to group them at this stage.